tests/test_fuzzy_checker.py

```diff
@@ -1,7 +1,247 @@
+            assert i > 1_100
+            break
+        if i > 1_300:
+            raise RuntimeError("Test did not reach the expected warning")
+        i += 1
+
+
+@pytest.mark.parametrize(
+    "numerator",
+    [
+        1_008,
+        10_008,
+        100_008,
+        1_000_008,
+    ],
+)
+@pytest.mark.parametrize("denominator", OBSERVED_DENOMINATORS)
+def test__calculate_bayes_factor(numerator: int, denominator: int) -> None:
+    if numerator > denominator:
+        pytest.skip("Numerator cannot be greater than denominator.")
+    # Parametrize rv_discrete for no bug distribution
+    # I am keeping the defaults for the bug distribution to remain 0.5 for alpha and beta
+    bug_issue_distribution = scipy.stats.betabinom(a=0.5, b=0.5, n=denominator)
+    no_bug_issue_distribution = scipy.stats.binom(p=TARGET_PROPORTION, n=denominator)
+    bayes_factor = FuzzyChecker()._calculate_bayes_factor(
+        numerator, bug_issue_distribution, no_bug_issue_distribution
+    )
+    assert isinstance(bayes_factor, float)
+    assert bayes_factor > 0
+
+
+def test_zero_division__calculate_bayes_factor(caplog: LogCaptureFixture) -> None:
+    # This is just testing that we will hit a zero division error or floating point error
+    # and handle it correctly.
+    # We want the case where we observe a proportion that indicates an event is very likely
+    # but we expect it to be very unlikely.
+    numerator = 10_000_000
+    denominator = 10_000_000 - 1
+    target_proportion = 0.1
+    # I am keeping the defaults for the bug distribution to remain 0.5 for alpha and beta
+    bug_issue_distribution = scipy.stats.betabinom(a=0.5, b=0.5, n=denominator)
+    no_bug_issue_distribution = scipy.stats.binom(p=target_proportion, n=denominator)
+    bayes_factor = FuzzyChecker()._calculate_bayes_factor(
+        numerator, bug_issue_distribution, no_bug_issue_distribution
+    )
+    assert bayes_factor == float(np.finfo(float).max)
+
+
+@pytest.mark.parametrize("lower_bound", LOWER_BOUNDS, ids=lambda x: x)
+@pytest.mark.parametrize("width", WIDTHS, ids=lambda x: x)
+def test__fit_beta_distribution_to_uncertainty_interval(
+    lower_bound: float, width: float
+) -> None:
+    upper_bound = lower_bound + width
+    if upper_bound >= 1:
+        pytest.skip("Upper bound cannot be more than 1.")
+    a, b = FuzzyChecker()._fit_beta_distribution_to_uncertainty_interval(
+        lower_bound, upper_bound
+    )
+    dist = scipy.stats.beta(
+        a=a,
+        b=b,
+    )
+    with np.errstate(under="ignore"):
+        lb_cdf = dist.cdf(lower_bound)
+        ub_cdf = dist.cdf(upper_bound)
+    assert np.isclose(
+        lb_cdf, 0.025, atol=0.01
+    ), f"{lb_cdf} not close to {0.025}, {lower_bound} {upper_bound}"
+    assert np.isclose(
+        ub_cdf, 0.975, atol=0.01
+    ), f"{ub_cdf} not close to {0.975}, {lower_bound} {upper_bound}"
+
+
+def test__no_best_fit_beta_distribution(caplog: LogCaptureFixture) -> None:
```

Comment:
I don't like this test very much. We should just choose one or two cases that we _know_ will be very hard to fit to, instead of iterating all of the cases.

```suggestion
def test__imprecise_fit_beta_distribution(caplog: LogCaptureFixture) -> None:
```